lib/ddtrace/span_operation.rb

```diff
@@ -189,6 +182,22 @@ def finished?
       span.stopped?
     end
 
+    # Set this span's parent, inheriting any properties not explicitly set.
+    # If the parent is nil, set the span as the root span.
+    def parent=(parent)
```

Comment:
Important: I just noticed that `parent=` changes state on both the `SpanOperation` and the `Span`; but `parent_id=` only changes the `Span` and will leave the `SpanOperation` in an inconsistent state, because `@parent` will be incorrect.

Reply:
Yeah, that one the reasons we are moving to get rid of `parent`.

There's also the fact that for distributed spans (the ones that have a parent outside the of the current application), we can only ever set the `parent_id` never the `parent` because a parent Span object doesn't exist locally.

This PR doesn't remove `parent` completely, but that's the final goal.